### ISSUE TITLE
[Feature?] What about cache ?

### ISSUE THREAD

**Issue by petyots** [Laravel-Backpack/CRUD#364]
Hello, first of all i want to say that the project is just AWESOME  !!! :+1: But i can't find any events for cache on add, on update, on delete ... So my question is do you plan to support some of the cache drivers ?

**Comment by IvanBernatovic** [comment]
Can you elaborate more on this? Some use cases would be helpful.

**Comment by petyots** [comment]
For example : We have BackPack/News module but there is no events when you CRUD - it, I mean when you add new article you should update the cache for performance. It can be done straight in the vendor package Article Model, but i think it will be easier for a lot of people to be built it so we can listen for the events and do our logic.

**Comment by OwenMelbz** [comment]
Just use laravels built in eloquent observers?

https://laravel.com/docs/5.3/eloquent#observers

You can also add code after the store/update methods

e.g

```
    public function store(StoreRequest $request)
    {
        $return = parent::storeCrud();
        // do your custom code here
        return $return;
    }

    public function update(UpdateRequest $request)
    {
        $return = parent::updateCrud();
        // do your custom code here
        return $return;
    }
```

Bearing in mind laravel doesn't cache anything but views by default - you're required to handle this cache yourself.

Backpack doesn't know/handle what your caching system is, so it won't do anything with one.

But you can use provided functionality to handle it yourself (after all you added the caching in yourself) 

TL;DR

Either use the store/update methods in the controller to do your customised logic, or use eloquent observers.

**Comment by petyots** [comment]
That's what I am saying we have to do this at the vendor package, it will be better if backpack support memcached or file or redis...

**Comment by OwenMelbz** [comment]
It does "support" it.

You still have to add it in yourself in your own controllers.

Backpack is a CRUD framework, not an Application Framework - that part is Laravel.

You will need to clear your own cache, with what ever system you're using.

Maybe if you share your code so people can see the problem it will make sense.

But so far from the information we've got, the answer is more than likely "No" as its nothing to do with Backpack

**Comment by petyots** [comment]
Alright :) Thanks for the info it's a kind of misunderstanding... Anyway thanks ppl :)
